app/src/main/java/org/fossasia/openevent/general/search/SearchLocationFragment.kt

```diff
@@ -57,61 +61,73 @@ class SearchLocationFragment : Fragment() {
         val permission =
-            ContextCompat.checkSelfPermission(requireContext(), Manifest.permission.ACCESS_COARSE_LOCATION)
+            ContextCompat.checkSelfPermission(requireContext(),
+                Manifest.permission.ACCESS_COARSE_LOCATION)
         if (permission != PackageManager.PERMISSION_GRANTED) {
             requestPermissions(arrayOf(Manifest.permission.ACCESS_COARSE_LOCATION,
                 Manifest.permission.ACCESS_FINE_LOCATION), LOCATION_PERMISSION_REQUEST)
         }
     }
 
     private fun redirectToMain() {
-        val fragmentId = if (safeArgs.fromSearchFragment) R.id.searchFragment else R.id.eventsFragment
+        val fragmentId =
+            if (safeArgs.fromSearchFragment) R.id.searchFragment else R.id.eventsFragment
         Navigation.findNavController(rootView).popBackStack(fragmentId, false)
     }
 
-    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
+    private fun setupPlaceAutoCompleteFrag(savedInstanceState: Bundle?) {
+
+        val autocompleteFragment: PlaceAutocompleteFragment?
+        if (savedInstanceState == null) {
+            val placeOptions = PlaceOptions.builder().build(PlaceOptions.MODE_CARDS)
+            autocompleteFragment = PlaceAutocompleteFragment.newInstance(
+                BuildConfig.MAPBOX_KEY, placeOptions)
+            val transaction = fragmentManager?.beginTransaction()
+            transaction?.add(R.id.autocomplete_frag_container, autocompleteFragment,
+                AUTOCOMPLETE_FRAG_TAG)
+            transaction?.commit()
+        } else {
+            autocompleteFragment = fragmentManager?.findFragmentByTag(
+                AUTOCOMPLETE_FRAG_TAG) as? PlaceAutocompleteFragment
+        }
+
+        autocompleteFragment?.setOnPlaceSelectedListener(object : PlaceSelectionListener {
+            override fun onPlaceSelected(carmenFeature: CarmenFeature) {
+                val location = carmenFeature.placeName()?.split(",")?.first()
+                location?.let {
+                    searchLocationViewModel.saveSearch(it)
+                    redirectToMain()
+                }
+            }
+
+            override fun onCancel() {
+                activity?.onBackPressed()
+            }
+        })
+    }
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int,
+        permissions: Array<String>,
+        grantResults: IntArray
+    ) {
         when (requestCode) {
             LOCATION_PERMISSION_REQUEST -> {
                 if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                     geoLocationViewModel.configure()
                 } else {
-                    Snackbar.make(rootView, R.string.cannot_fetch_location, Snackbar.LENGTH_SHORT).show()
+                    Snackbar.make(rootView, R.string.cannot_fetch_location, Snackbar.LENGTH_SHORT)
+                        .show()
```

Comment:
Unnecessary change

Reply:
I just ran Code Format , it did that automatically.

Reply:
Don't run auto format in PRs.

Reply:
Do I have to revert this change ?

Reply:
Yes

Reply:
ok will do it

Reply:
@ShridharGoel  I have removed unnecessary changes . Please merge this